### PR TITLE
Add workflow to push the release

### PR DESCRIPTION
**Pull request by therve**
SSIA.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -1,34 +1,42 @@
...
+          SHA: $${ github.event.pull_request.head.sha }}
         with:
-          python-version: "3.10"
-
-      - name: Install datadog_checks_dev
-        run: |
-          python -m pip install datadog_checks_dev[cli]
+          github-token: ${{ steps.get_token.outputs.token }}
+          script: |
+            const tagName = process.env.RELEASE_TAG.split("/")[1];
```
Comment (skarimo) [inline .github/workflows/release.yml:28]: Im not too familiar with how github actions work but is the `RELEASE_TAG` env var shared between the workflows somehow? Also why do we split it in this case?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -1,34 +1,42 @@
...
+        id: get_token
+        uses: tibdex/github-app-token@v1
         with:
-          fetch-depth: 0
-
-      - uses: actions/setup-python@v2
-        name: Install Python
+          app_id: ${{ secrets.PIPELINE_GITHUB_APP_ID }}
+          private_key: ${{ secrets.PIPELINE_GITHUB_APP_PRIVATE_KEY }}
+      - name: Create release
```
Comment (skarimo) [inline .github/workflows/release.yml:21]:
```suggestion
      - name: Create release
        uses: actions/github-script@v5
```

File: .github/workflows/release.yml (hunk 1 above)
Comment (therve) [reply .github/workflows/release.yml:28]: Missed the variable name, should be `RELEASE_BRANCH`.

File: .github/workflows/release.yml (hunk 2 above)
Comment (therve) [reply .github/workflows/release.yml:21]: Thanks!

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    types:
+      - published
+
+jobs:
+  upload_release:
+    name: Upload release
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+        ref: $${{ github.event.release.tag_name }}
```
Comment (jirikuncar) [inline .github/workflows/publish.yml:14]:
```suggestion
        ref: ${{ github.event.release.tag_name }}
```

File: .github/workflows/publish.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+          python -m pip install datadog_checks_dev[cli]
+
+      - name: Set ddev pypi credentials
+        run: |
+          ddev config set pypi.user __token__
+          ddev config set pypi.pass ${{ secrets.PYPI_TOKEN }}
+
+      - name: Publish the wheel to PyPI
+        run: |
+          ddev release upload .
```
Comment (jirikuncar) [inline .github/workflows/publish.yml:35]: https://github.com/pypa/gh-action-pypi-publish WDYT?

**Review by jirikuncar — APPROVED** [review]

**Pull request merged**